### PR TITLE
Simplify `pack_frames_prelude`

### PR DESCRIPTION
**Pull request by jakirkham**
Reduce to a single call of `struct.pack`.

### CODE REVIEW

File: distributed/protocol/utils.py [hunk 1]
```diff
@@ -89,10 +89,7 @@ def merge_frames(header, frames):
 
 
 def pack_frames_prelude(frames):
-    lengths = [struct.pack("Q", len(frames))] + [
-        struct.pack("Q", nbytes(frame)) for frame in frames
-    ]
-    return b"".join(lengths)
+    return struct.pack(f"Q{len(frames)}Q", len(frames), *map(nbytes, frames))
```
Comment (quasiben) [inline distributed/protocol/utils.py:92]: With more complicated comprehensions I sometimes have trouble reading what it does .  Maybe include a docstring like:
> Pack frames into bytestring of length of frames and size of each frame

Maybe it's fine...if you thinks it's clear then merge away

Comment (jakirkham) [reply distributed/protocol/utils.py]: Assigned a couple variables beforehand. Please let me know if that is clearer.

Comment (quasiben) [reply distributed/protocol/utils.py]: Thanks @jakirkham

**Comment by quasiben** [comment]
Feel free to merge after CI passes

**Comment by jakirkham** [comment]
Thanks Ben! 😄

**Pull request merged**